django/core/paginator.py

```diff
@@ -52,6 +53,9 @@ def page(self, number):
         top = bottom + self.per_page
         if top + self.orphans >= self.count:
             top = self.count
+
+        if hasattr(self.object_list, 'ordered') and not self.object_list.ordered:
+            warnings.warn("Inconsistent pagination when no ordering is specified", RuntimeWarning)
```

Comment:
Actually it might be worth using a subclass of `DjangoRuntimeWarning` just like we do with `django.core.cache.base.CacheKeyWarning`.